src/app/Library/CrudPanel/Traits/SaveActions.php

```diff
@@ -15,52 +19,260 @@ public function getSaveActionDefaultForCurrentOperation()
+    {
+        if (is_array($saveAction)) {
+            //check for some mandatory fields
+            $saveAction['name'] ?? abort(500, 'Please define save action name.');
+            $saveAction['redirect'] = $saveAction['redirect'] ?? function ($crud, $request, $itemId) {
+                return $request->has('http_referrer') ? $request->get('http_referrer') : $crud->route;
+            };
+            $saveAction['permissions'] = $saveAction['permissions'] ?? true;
+            $saveAction['button_text'] = $saveAction['button_text'] ?? $saveAction['name'];
+            $saveAction['order'] = isset($saveAction['order']) ? $this->reorderSaveActions($saveAction['order']) : count($this->availableSaveActions) + 1;
+            $this->availableSaveActions[$saveAction['name']] = [];
+            $this->availableSaveActions[$saveAction['name']] = $saveAction;
+        }
+    }
+
+    /**
+     * Replaces setting order or forces some default.
+     *
+     * @param int $wantedOrder
+     * @return int
+     */
+    public function reorderSaveActions($wantedOrder)
+    {
+        if (! empty($this->availableSaveActions)) {
+            $lastOrder = max(array_column($this->availableSaveActions, 'order'));
+            foreach ($this->availableSaveActions as &$sv) {
+                if ($wantedOrder == $sv['order']) {
+                    if (empty(array_filter($this->availableSaveActions, function ($arr) {
+                        $arr['order'] == 1;
+                    }))) {
+                        $sv['order'] = 1;
+                    } else {
+                        $sv['order'] = $lastOrder + 1;
+                    }
+                }
+            }
+        }
+
+        return $wantedOrder;
+    }
+
+    /**
+     * Replace the current save actions with the ones provided.
+     *
+     * @param array $saveActions
+     * @return void
+     */
+    public function replaceSaveActions($saveActions)
+    {
+        if (is_array($saveActions)) {
+            //we reset all save actions
+            $this->availableSaveActions = [];
+
+            $this->applyRegisterSaveActions($saveActions);
+        }
+    }
+
+    /**
+     * Allow the developer to unset save actions.
+     *
+     * @param string $saveAction
+     * @return void
+     */
+    public function forgetSaveActions($saveAction)
+    {
+        if (is_array($saveAction)) {
+            foreach ($saveAction as $sv) {
+                if (isset($this->availableSaveActions[$sv])) {
+                    unset($this->availableSaveActions[$sv]);
+                }
+            }
+        } else {
+            if (isset($this->availableSaveActions[$saveAction])) {
+                unset($this->availableSaveActions[$saveAction]);
+            }
+        }
+    }
+
+    /**
+     * Allow the developer to unset all save actions.
+     *
+     * @param string $saveAction
+     * @return void
+     */
+    public function forgetAllSaveActions()
```

Comment:
I would rename this to ```removeAllSaveActions()``` for consistency with the API for Fields, Columns, Buttons, etc.

Reply:
:+1: